packages/core/core/src/services/document-service/document-engine.ts

```diff
@@ -56,44 +62,59 @@ const createDocumentEngine = ({
   async findMany(uid, params) {
     const { kind } = strapi.getModel(uid);
 
-    const query = transformParamsToQuery(uid, params || ({} as any));
-    query.where = { ...params?.lookup, ...query.where };
+    const query = await pipeAsync(
+      (params) => transformParamsDocumentId(uid, params, { isDraft: true, locale: params.locale }),
+      (params) => transformParamsToQuery(uid, params),
+      (query) => set('where', { ...params?.lookup, ...query.where }, query)
+    )(params || {});
 
     if (kind === 'singleType') {
       return db.query(uid).findOne(query);
```

Comment:
Do we not need to transform the output for single types too?

Reply:
woops, yes we neeed it!